README.md

```diff
@@ -1,4 +1,241 @@
+        name TEXT, 
+        album TEXT
+        )
+        SQL
+    DB[:conn].execute(sql) 
+  end
+
+end
+```
+
+Let's break down this code.
+
+#### The `id` Attribute
+
+Notice that we are initializing an individual `Song` instance with an `id` attribute that has a default value of `nil`. Why are we doing this? First of all, songs need an `id` attribute only because they will saved into the database and we know that each table row needs an `id` value which is the primary key. 
+
+When we create a new song with the `Song.new` method, we *do not set that song's id*. A song gets an `id` only when it gets saved into the database (more on inserting songs into the database later). We therefore set the default value of the `id` argument that the `#initialize` method takes equal to `nil`, so that we can create new song instances that *do not have an `id` value. We'll leave that up to the database to handle later on. 
+
+Similarly, we do not have an `attr_accessor` for `id`, because we never want to set or change a song's `id` manually. The database alone is responsible for setting a song's `id`. We'll learn how later on. 
+
+#### The `#create_table` Method
+
+Above, we created a class method, `#create_table`, that crafts a SQL statement to create a songs table and give that table column names that match the attributes of an individual instance of `Song`. Why is the `#create_table` method a class method? Well, it is *not* the responsibility of an individual song to create the table it will eventually be saved into. It is the job of the class as a whole to create the table that it is mapped to. 
+
+**Top-Tip:** For strings that will take up multiple lines in your text editor, use a [heredoc](https://en.wikipedia.org/wiki/Here_document) to create a string that runs on to multiple lines. To create a heredoc, we use:
+
+`<<-` + `name of language contained in our multiline statement` + `the string, on multiple lines` + `name of language`. 
+
+You don't have to use a heredoc, it's just a helpful too for crafting long strings in Ruby. Back to our regularly scheduled programming...
+
+Now that our songs table exists, we can learn how to save data regarding individual songs into that table. 
+
+## Mapping Class Instances to Table Rows: Writing a `#save` method to `INSERT` data into the table
+
+When we say that we are saving data to our database, what data are we referring to? If individual instances of a class are "mapped" to rows in  a table, does that mean that the instances themselves, these individual Ruby objects, are saved into the database?
+
+Actually, **we are not saving Ruby objects in our database.** We are going to take the individual attributes of a given instance, in this case a song's name and album, and save *those attributes that describe an individual song* to the database, as one, single row.
+
+Let's build an instance method, `#save`, that saves a given instance of our `Song` class into the songs table of our database:
```

Comment:
Before showing them the "abstracted" version show them the concerte. If you had Hot Line Bling and Golddigger and you wanted to insert both, what would the SQL command be. OK what of that is the same, and what is variable. Now we can build a method